### DISCUSSION TITLE
Using media API

### DISCUSSION THREAD

**Discussion by mbronstein1** [blakeblackshear/frigate#15314] [Ask A Question]
I've noticed that when viewing media in the dashboard, the api used is the vod with specific start and end times, even tho there is an event_id-based route (in addition to many others). I've noticed on my end, using the event_id route (either w/ mp4 or m3u8), there are far more likely to be corrupted videos or 404's. I'll even see that the video for the event DOES exist in the dashboard (using m3u8 and timestamps), but via the event id routes it returns 404.

My questions are:
1. What would lead the api to return a 404 when the event id route is hit but return a 200 (or 206 for partial) when the timestamps are?
   a. What could occur during an event to make the recording be thrown out w/ relation to the event but still be present in the recordings api?
3. Is the timestamps approach generally more reliable (seemingly it is if that is the main route the Frigate UI uses)?

Thanks for your time! I'm just trying to better understand these so I know how to reduce the number of 404s based on what is actually saved in the file system (obviously this is also based on retention but I'm getting 404s sometimes on events that just occurred).

**Comment by NickM-27** [comment]
Where are you getting the event ID? Currently in 0.14 the UI works off of review items which are different from events (hence why the UI uses the time based m3u8 APIs)

**Reply by mbronstein1** [reply]
I'm getting the event id from the events mqtt message. Originally, I was just using the clips.mp4 route (w/ 0.13.2) which took the event id as a path parameter. But w/ 0.14+ (I'm currently using 0.15 beta 2), I noticed there was an event id m3u8 route. However, with the inconsistencies, I decided to look at how the Frigate UI handles recordings and it seems you always use the time based m3u8. I was wondering if that is just more reliable in general based on how recordings are being copied even though the event_id and the time stamp would be from the same mqtt event, and (ideally) return the same video.

**Reply by NickM-27** [reply]
Yes in general that is the better API to use
